packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybrid2GL.hpp

```diff
@@ -0,0 +1,1532 @@
+      if(verbose) std::cout<<comm->getRank()<<": constructing rand and GIDs views\n";
+      Kokkos::View<int*, device_type> rand_dev("Random View", rand.size());
+      typename Kokkos::View<int*, device_type>::HostMirror rand_host = Kokkos::create_mirror(rand_dev);
+      for(size_t i = 0; i < rand.size(); i ++) rand_host(i) = rand[i];
+      Kokkos::deep_copy(rand_dev,rand_host);
+      
+      //create a view for the global IDs of each vertex.
+      Kokkos::View<gno_t*, device_type> gid_dev("GIDs", gids.size());
+      typename Kokkos::View<gno_t*, device_type>::HostMirror gid_host = Kokkos::create_mirror(gid_dev);
+      for(size_t i = 0; i < gids.size(); i++) gid_host(i) = gids[i];
+      Kokkos::deep_copy(gid_dev,gid_host);
+       
+      //These variables will be initialized by constructBoundary()
+      //
+      //boundary_verts_dev holds all owned vertices that could possibly conflict
+      //with a remote vertex. Stores LIDs.
+      Kokkos::View<lno_t*, device_type> boundary_verts_dev; 
+      //this is the number of vertices in the boundary.
+      gno_t boundary_size = 0;
+      if(verbose) std::cout<<comm->getRank()<<": constructing communication and recoloring lists\n";
+      
+      //We keep track of the vertices that need to get recolored
+      //this list can contain ghost vertices, so it needs to be initialized
+      //to the number of all vertices on the local process.
+      //rand has an entry for each vertex, so its size corresponds to what is needed.
+      Kokkos::View<lno_t*, device_type> verts_to_recolor_view("verts to recolor", rand.size());
+      typename Kokkos::View<lno_t*, device_type>::HostMirror verts_to_recolor_host = create_mirror(verts_to_recolor_view);
+      Kokkos::View<lno_t*, device_type, Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_recolor_atomic = verts_to_recolor_view;
+      
+      //This view keeps track of the size of the list of vertices to recolor.
+      Kokkos::View<int*, device_type> verts_to_recolor_size("verts to recolor size",1);
+      Kokkos::View<int*, device_type, Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_recolor_size_atomic = verts_to_recolor_size;
+      typename Kokkos::View<int*, device_type>::HostMirror verts_to_recolor_size_host = create_mirror(verts_to_recolor_size);
+      
+      //initialize the host view
+      verts_to_recolor_size_host(0) = 0;
+      //copy to device
+      Kokkos::deep_copy(verts_to_recolor_size, verts_to_recolor_size_host);
+
+      //verts to send can only include locally owned vertices,
+      //so we can safely allocate the view to size n_local.
+      //
+      //This view is a list of local vertices that are going to be
+      //recolored, and need to be sent to their remote copies.
+      Kokkos::View<lno_t*, device_type> verts_to_send_view("verts to send", n_local);
+      Kokkos::View<lno_t*, device_type, Kokkos::MemoryTraits<Kokkos::Atomic>> verts_to_send_atomic = verts_to_send_view;
```

Comment:
Why is this atomic view needed? How will this view be used?

Reply:
This is not needed, I have removed it